model/preparation.go

```diff
@@ -11,6 +11,13 @@ import (
 	"gorm.io/gorm"
 )
 
+type PieceType string
+
+const (
+	DataPiece PieceType = "data"
+	DagPiece  PieceType = "dag"
```

Comment:
Can you document the difference between these two? Data and DAG are both very overloaded terms in this space.